versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/codecs/BsonToEntityConverter.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.mongodb.codecs;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.bson.BsonReader;
+import org.bson.BsonType;
+
+import com.dremio.nessie.versioned.store.Entity;
+import com.dremio.nessie.versioned.store.mongodb.MongoDbConstants;
+
+/**
+ * Converter for serializing a BSON object to an Entity.
+ */
+public class BsonToEntityConverter {
```

Comment:
Why not put this in the codec factory as a private class?

Also, rather than use static methods, please put the body of the operation in a class instance and then use a static singleton instance.